e2e/nomostest/helm/helm.go

```diff
@@ -0,0 +1,177 @@
+	loginCmd.Stdin, err = authCmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to setup command pipe: %v", err)
+	}
+	if err := loginCmd.Start(); err != nil {
+		return fmt.Errorf("failed to start login command: %v", err)
+	}
+	if err := authCmd.Run(); err != nil {
+		return fmt.Errorf("failed to run auth command: %v", err)
+	}
+	if err := loginCmd.Wait(); err != nil {
+		return fmt.Errorf("failed to wait for login command: %v", err)
+	}
+	return nil
+}
+
+// CopyChartFromLocal accepts a local path to a helm chart and recursively copies it to r.Dir, modifying
+// the name of the copied chart from its original name to r.ChartName
+func (r *RemoteHelmChart) CopyChartFromLocal(chartPath, originalChartName string) error {
+	if err := copyutil.CopyDir(chartPath, r.Dir); err != nil {
+		return fmt.Errorf("failed to copy helm chart: %v", err)
+	}
+	if err := findAndReplaceInFile(filepath.Join(r.Dir, "Chart.yaml"), fmt.Sprintf("name: %s", originalChartName), fmt.Sprintf("name: %s", r.ChartName)); err != nil {
+		return fmt.Errorf("failed to rename helm chart: %v", err)
+	}
+	return nil
+}
+
+// Push will package and push the helm chart located at r.Dir to the remote registry.
+func (r *RemoteHelmChart) Push() error {
+	if _, err := r.Shell.Helm("package", r.Dir, "--destination", r.Dir); err != nil {
+		return fmt.Errorf("failed to package helm chart: %v", err)
+	}
+	chartFile := filepath.Join(r.Dir, fmt.Sprintf("%s-%s.tgz", r.ChartName, r.ChartVersion))
+	if out, err := r.Shell.Helm("push", chartFile, r.Registry); err != nil {
+		return fmt.Errorf("failed to run `helm push`: %s; %v", string(out), err)
+	}
+	return nil
+}
+
+// PushHelmChart pushes a new helm chart for use during an e2e test. Returns the name of the chart that gets pushed and any errors that are encountered.
+func PushHelmChart(nt *nomostest.NT, helmchart, version string) (string, error) {
+	nt.T.Log("Push helm chart to the artifact registry")
+
+	chartName := generateChartName(helmchart)
+	nt.T.Cleanup(func() {
+		if err := cleanHelmImages(nt, chartName); err != nil {
+			nt.T.Errorf(err.Error())
+		}
+	})
+
+	remoteHelmChart := NewRemoteHelmChart(nt.Shell, PrivateARHelmHost, PrivateARHelmRegistry, nt.TmpDir, chartName, version)
+	err := remoteHelmChart.CopyChartFromLocal(fmt.Sprintf("../testdata/helm-charts/%s", helmchart), helmchart)
+	if err != nil {
+		return "", err
+	}
+	if err := remoteHelmChart.RegistryLogin(); err != nil {
+		return "", fmt.Errorf("failed to login to the helm registry: %v", err)
+	}
+	if err := remoteHelmChart.Push(); err != nil {
+		return "", err
+	}
+
+	return chartName, nil
+}
+
+// creates a chart name from current project id, cluster name, and timestamp
+func generateChartName(helmchart string) string {
+	chartName := fmt.Sprintf("%s-%s-%s-%s", helmchart, nomostesting.GCPProjectIDFromEnv, nomostesting.GCPClusterFromEnv, timestampAsString())
```

Comment:
Probably don't need the GCP project in the name, given the artifact registry is already scoped to the project. Might help with truncation if this is omitted

Reply:
> Might help with truncation if this is omitted 

Thanks! That does help a lot. Done.